backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/orchestration/impl/ElasticsearchStubWorkExecutionContext.java

```diff
@@ -24,4 +26,14 @@ public ElasticsearchStubWorkExecutionContext(ElasticsearchClient client) {
 	public ElasticsearchClient getClient() {
 		return client;
 	}
+
+	@Override
+	public void setIndexDirty(URLEncodedString indexName) {
+		// TODO scope execution context per request
+	}
+
+	@Override
+	public GsonProvider getGsonProvider() {
+		return client.getGsonProvider();
```

Comment:
Having more lines of code for something that could change in the future, from my experience could be a premature optimization. On the other hand the impact here is not very critical, following the collaboration I would have +10 -5 LOC change. So I will change ;)

Reply:
Thanks. In general it may be true, but here we're talking about controlling the flow of information, the responsibilities of each class, and encapsulation in general. So I wouldn't call that "optimization", it's just design. Adding a getter here and there may be easier in the short term, but over time it introduces dependencies between components and makes changes harder (because they have a larger impact on the codebase).

Anyway, in this particular case I'm working on a PR where, indeed, I need to introduce another implementation of the Elasticsearch client for tests. So, rest assured it's not for nothing.